src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -144,13 +145,27 @@ object ResourceMatcher extends StrictLogging {
     // that means if the resources that are matched are still unreserved.
     def needToReserveDisk = selector.needToReserve && runSpec.diskForPersistentVolumes > 0
 
-    val diskMatch = if (needToReserveDisk)
+    val diskMatch = if (needToReserveDisk) {
+      val volumes = runSpec.persistentVolumes
+      val mounts = runSpec.persistentVolumeMounts
+      val (namedVolumes, namedMounts) =
+        if (volumes.exists(_.name.isEmpty) && volumes.length == mounts.length) {
```

Comment:
I`m not sure if I can follow this lines...what is the intention to behind this condition above and the separate logic below?

Reply:
I will elaborate on it.